src/Search/Search.js

```diff
@@ -1,19 +1,70 @@
 // @flow
-import React, {useState} from "react";
-import { useO2xpProvider } from "../O2xpContext/O2xpContext";
+
+import React, { useState, useEffect } from "react";
+import cloneDeep from "lodash/cloneDeep";
+import IconButton from "@material-ui/core/IconButton";
+import SearchIcon from "@material-ui/icons/Search";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import Autocomplete from "@material-ui/lab/Autocomplete";
+import Switch from "@material-ui/core/Switch";
+import HelpIcon from "@material-ui/icons/Help";
+import Tooltip from "@material-ui/core/Tooltip";
+import TextField from "@material-ui/core/TextField";
+import useO2xpProvider from "../hooks/useO2xpProvider";
+import "../style.css";
+import { simpleSearch, transformString, managePrioritiesQueries } from "./searchTools";
 
 const Search = () => {
   const {
-    state: { data },
+    state: { data, columns },
     dispatch
   } = useO2xpProvider();
 
+  const [open, setOpen] = useState(true);
   const [filterBy, setFilterBy] = useState("");
+  const [queryMode, setQueryMode] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [colorChange, setColorChange] = useState("primary");
+  const [placeholderQueryMode, setPlaceholderQueryMode] = useState("Column = Value");
+
+  useEffect(() => {
+    const newPlaceholderQueryMode: string = queryMode
+      ? "Column = Value"
+      : "Simple search";
+    setPlaceholderQueryMode(newPlaceholderQueryMode);
+  }, [queryMode]);
 
-  const handleChange = e => {
+  useEffect(() => {
+    const newColorChange: string = filterBy.length === 0 ? "primary" : "secondary";
+    setColorChange(newColorChange);
+  }, [filterBy]);
+
+  const handleChange = (e: SyntheticInputEvent<*>) => {
     const { value } = e.target;
-    const { baseRowsData } = data;
-    const newRowsData = baseRowsData.filter(el => el.name.startsWith(value));
+    const { baseRowsData }: { rowsData: Object[], baseRowsData: Object[] } = data;
+    const clBaseRowsData: Object[] = cloneDeep(baseRowsData);
```

Comment:
Why are you assigning valu to this `clBaseRowsData` and then to `newRowsData` ? 
 `clBaseRowsData`  is not necessary